tests/dataset/test_export_datasets_and_create_metadata_db.py

```diff
@@ -0,0 +1,483 @@
+        target_db_path = Path(temp_dir) / "target.db"
+        export_path = Path(temp_dir) / "exports"
+        
+        # Create source database
+        source_conn = connect(source_db_path)
+        exp = load_or_create_experiment(
+            experiment_name="test_exp",
+            sample_name="test_sample",
+            conn=source_conn
+        )
+        
+        # Create interdependencies
+        x = ParamSpec("x", "numeric", unit="V")
+        y = ParamSpec("y", "numeric", unit="A")
+        interdeps = InterDependencies_(dependencies={y: (x,)})
+        
+        created_datasets = []
+        
+        # Create several datasets with different characteristics
+        for i in range(5):
+            dataset = DataSet(conn=source_conn, exp_id=exp.exp_id)
+            dataset.set_interdependencies(interdeps)
+            dataset.mark_started()
+            
+            # Add varying amounts of data
+            for j in range(i + 1):  # 1, 2, 3, 4, 5 data points respectively
+                dataset.add_results([{"x": j, "y": j * (i + 1)}])
+            
+            if i < 4:  # Leave one dataset incomplete
+                dataset.mark_completed()
+            
+            created_datasets.append(dataset)
+        
+        source_conn.close()
+        
+        # Run the export function
+        result = export_datasets_and_create_metadata_db(
+            source_db_path=source_db_path,
+            target_db_path=target_db_path,
+            export_path=export_path,
+        )
+        
+        # Check that all datasets were processed
+        assert len(result) == 5
+        
+        # Check that target database has all runs
+        target_conn = connect(target_db_path)
+        target_runs = get_runs(target_conn)
+        target_conn.close()
+        
+        assert len(target_runs) == 5
+        
+        # The incomplete dataset should be copied as-is
+        incomplete_dataset = created_datasets[-1]
+        assert result[incomplete_dataset.run_id] == "copied_as_is"
+
+
+def test_export_datasets_status_reporting():
+    """Test that the function returns detailed status information"""
+    with tempfile.TemporaryDirectory() as temp_dir:
+        source_db_path = Path(temp_dir) / "source.db"
+        target_db_path = Path(temp_dir) / "target.db"
+        export_path = Path(temp_dir) / "exports"
+        
+        # Create source database with a completed dataset
+        source_conn = connect(source_db_path)
+        exp = load_or_create_experiment(
+            experiment_name="test_exp",
+            sample_name="test_sample",
+            conn=source_conn
+        )
+        
+        # Create interdependencies
+        x = ParamSpec("x", "numeric", unit="V")
+        y = ParamSpec("y", "numeric", unit="A")
+        interdeps = InterDependencies_(dependencies={y: (x,)})
+        
+        # Create and complete a dataset
+        dataset = DataSet(conn=source_conn, exp_id=exp.exp_id)
```

Comment:
perhaps refactor dataset creation into a separate pytest fixture that can be re-used across the tests?